File: 554_Brick_Wall.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        unordered_map<int, int> edges;
        int size = wall.size(), min_bricks = size;
        for (int i = 0; i < size; i++) {
            for (int j = 0, width = 0; j < wall[i].size() - 1; j++) {
                min_bricks = min(min_bricks, size - (++edges[width += wall[i][j]]));
            }
        }
        return min_bricks;
    }
};

File: 554_Brick_Wall.py
class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        n = len(wall)
        edges = collections.defaultdict(int)
        minimum = n
        for bricks in wall:
            width = 0
            for brick in bricks[:-1]:
                width += brick
                edges[width] += 1
                minimum = min(minimum, n - edges[width])
        return minimum
